### PR TITLE
feat(docker): add command to inject packed docker images into firmware images

### PR DESCRIPTION
**Pull request by empwilli**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -64,16 +64,23 @@ uuid = { version = "0.8", default-features = false, features = ["v4"] }
 url = { version = "2.4" }
 validator = { version = "0.14", features = ["derive"] }
 xz2 = "0.1"
+tempfile = "3.10.1"
```
Comment (JanZachmann) [inline Cargo.toml:67]: pls. keep alphabetical order

File: src/cli.rs [hunk 2]
```diff
@@ -9,6 +9,31 @@ use url::Url;
...
+    Inject {
+        /// full qualified name of the docker image
+        #[clap(short = 'd', long = "docker-image", required(true))]
+        docker_image: String,
+        /// path to wic image file (optionally compressed with xz, bzip2 or gzip)
+        #[arg(short = 'i', long = "image")]
+        image: PathBuf,
+        /// partition to store the image to
+        #[clap(short = 'a', long = "partition", value_enum)]
+        partition: Partition,
```
Comment (JanZachmann) [inline src/cli.rs:25]: wouldn't it make sense to set factory here as the default?

File: src/cli.rs [hunk 3]
```diff
@@ -9,6 +9,31 @@ use url::Url;
...
+        /// file path to store the docker image to
+        #[clap(short = 'e', long = "dest")]
+        dest: PathBuf,
```
Comment (JanZachmann) [inline src/cli.rs:28]: wouldn't it make sense to set a default here as well?

File: src/docker.rs [hunk 4]
```diff
@@ -0,0 +1,105 @@
+use anyhow::{Context, Result};
+use std::io::Write;
+use std::path::PathBuf;
+
+use crate::image::Architecture;
+
+impl From<Architecture> for &str {
+    fn from(arch: Architecture) -> &'static str {
+        match arch {
+            Architecture::Arm64 => "linux/arm64",
```
Comment (JanZachmann) [inline src/docker.rs:10]: what about the other archs?

File: src/docker.rs [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+            .context("pull_docker_image: could not create compression stream")?;
+
+        let mut stream = xz2::write::XzEncoder::new_stream(destination, stream_builder);
+
+        std::io::copy(&mut source, &mut stream)
+            .context("compress_docker_image: could not compress docker file")?;
+
+        let writer = stream.finish()?;
+        Ok(writer.into_inner()?)
+    }
```
Comment (JanZachmann) [inline src/docker.rs:43]: isn't it possible to reuse https://github.com/omnect/omnect-cli/blob/main/src/file/compression.rs ?

File: src/image.rs [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+pub enum Architecture {
+    Arm64,
+}
+
+impl TryInto<Architecture> for &str {
+    type Error = anyhow::Error;
+
+    fn try_into(self) -> Result<Architecture> {
+        let arch = match self {
+            "aarch64" => Architecture::Arm64,
```
Comment (JanZachmann) [inline src/image.rs:29]: what about amd64?

**Review by JanZachmann — CHANGES_REQUESTED** [review]

File: src/cli.rs (hunk 3 above)
Comment (empwilli) [reply src/cli.rs:28]: Is there a sensible default? @JoergZeidler ?

Comment (JoergZeidler) [reply src/cli.rs:28]: default could be the root path of the partition

File: src/image.rs (hunk 6 above)
Comment (empwilli) [reply src/image.rs:29]: any other architecture that we're supporting?

File: src/cli.rs (hunk 3 above)
Comment (mlilien) [reply src/cli.rs]: imho a better default would be 
factory partition and "/oci_images" as default path.

File: src/cli.rs [hunk 7]
```diff
@@ -9,6 +9,31 @@ use url::Url;
...
+        #[clap(short = 'd', long = "docker-image", required(true))]
+        docker_image: String,
+        /// path to wic image file (optionally compressed with xz, bzip2 or gzip)
+        #[arg(short = 'i', long = "image")]
+        image: PathBuf,
+        /// partition to store the image to
+        #[clap(short = 'a', long = "partition", value_enum, default_value = "factory")]
+        partition: Partition,
+        /// file path to store the docker image to
+        #[clap(short = 'e', long = "dest", default_value = "/")]
```
Comment (JanZachmann) [inline src/cli.rs:27]: I agree with Marcel and would go for /oci_imges

File: src/file/functions.rs [hunk 8]
```diff
@@ -25,6 +27,17 @@ struct PartitionInfo {
     end: String,
 }
 
+impl Display for Partition {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Partition::boot => write!(f, "boot"),
+            Partition::rootA => write!(f, "rootA"),
```
Comment (JanZachmann) [inline src/file/functions.rs:34]: can you rm rootA here: https://github.com/omnect/omnect-cli/blob/3e4d5f8b221a08c56550ac2fd81433729b383562/src/file/functions.rs#L16

FYI: @mlilien

**Review by JanZachmann — CHANGES_REQUESTED** [review]

File: src/file/functions.rs (hunk 8 above)
Comment (mlilien) [reply src/file/functions.rs:34]: @JanZachmann  imho not a good idea ... we want to read files from rootA
(it doesn't make sense to write to rootA though)

**Review by mlilien — APPROVED** [review]

**Review by JoergZeidler — APPROVED** [review]

**Review by JanZachmann — APPROVED** [review]

**Pull request merged**